### PR TITLE
Hai 42 db for local dev

### PR DESCRIPTION
**Pull request by Silvast**
This is a new branch to replace HAI-42-postgis-for-local-development since changing the structures just messed up the conflicts.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,8 +25,6 @@ Manual build
 * services/hanke-service
     * Haitaton 2.0 Hanke service
 
-## How to compile, build and run
```
Comment (muhassin) [inline README.md:28]: Why removing the heading?

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hello/db/DbController.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    fun listTables(connection: Connection): ArrayList<String> {
+        var listOfTableNames: ArrayList<String> = arrayListOf()
+        val sql = "select * from pg_tables where schemaname='public';"
+        val response = connection.createStatement().executeQuery(sql)
+        while (response.next()) {
+            listOfTableNames.add(response.getString("tablename"))
+        }
+        logger.info { "query response:" + "$listOfTableNames" }
+        return listOfTableNames
```
Comment (muhassin) [inline services/hanke-service/src/main/kotlin/fi/hel/haitaton/hello/db/DbController.kt:47]: Usually database connections need to be closed, one way or another. If reusing a connection (which I don't think is happening), at least the resultset and statement should be closed. And those within finally-block.

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hello/db/DbController.kt [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    lateinit var dbConfigProperties: DbConfigProperties
+
+    @GetMapping("/tablenames")
+    fun getAllTables(): String {
+
+        val dbUrl: String = dbConfigProperties.appDatasourceUrl + "?" +
+            "user=" + dbConfigProperties.appDatasourceUsername +
+            "&password=" + dbConfigProperties.appDatasourcePassword
+
+        val con = DriverManager.getConnection(dbUrl)
```
Comment (muhassin) [inline services/hanke-service/src/main/kotlin/fi/hel/haitaton/hello/db/DbController.kt:30]: Is there a reason to use DriverManager and not DataSource?  From DriverManager docs:
_NOTE: The DataSource interface, new in the JDBC 2.0 API, provides another way to connect to a data source. The use of a DataSource object is the preferred means of connecting to a data source._
Afaik, that "new" is by now ancient history (DataSource came in java 1.4)
(Well, this is just an example API, so anything goes, I guess.)

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hello/db/DbController.kt [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    fun listTables(connection: Connection): ArrayList<String> {
+        var listOfTableNames: ArrayList<String> = arrayListOf()
+        val sql = "select * from pg_tables where schemaname='public';"
+        val response = connection.createStatement().executeQuery(sql)
+        while (response.next()) {
+            listOfTableNames.add(response.getString("tablename"))
+        }
+        logger.info { "query response:" + "$listOfTableNames" }
```
Comment (muhassin) [inline services/hanke-service/src/main/kotlin/fi/hel/haitaton/hello/db/DbController.kt:46]: "query response: $listOfTablenames" ?

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hello/db/DbController.kt [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    fun getAllTables(): String {
+
+        val dbUrl: String = dbConfigProperties.appDatasourceUrl + "?" +
+            "user=" + dbConfigProperties.appDatasourceUsername +
+            "&password=" + dbConfigProperties.appDatasourcePassword
+
+        val con = DriverManager.getConnection(dbUrl)
+
+        logger.info {
+            "Connected to database: " +
```
Comment (muhassin) [inline services/hanke-service/src/main/kotlin/fi/hel/haitaton/hello/db/DbController.kt:33]: "Connected to database: ${dbConfigProperties.appDatasourceUrl}" ?

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hello/db/DbController.kt [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Autowired
+    lateinit var dbConfigProperties: DbConfigProperties
+
+    @GetMapping("/tablenames")
+    fun getAllTables(): String {
+
+        val dbUrl: String = dbConfigProperties.appDatasourceUrl + "?" +
+            "user=" + dbConfigProperties.appDatasourceUsername +
+            "&password=" + dbConfigProperties.appDatasourcePassword
```
Comment (muhassin) [inline services/hanke-service/src/main/kotlin/fi/hel/haitaton/hello/db/DbController.kt:28]: "${dbConfigProperties.appDatasourceUrl}?user=${dbConfigProperties.appDatasourceUsername}&password=${dbConfigProperties.appDatasourcePassword}"  ?

File: README.md (hunk 1 above)
Comment (Silvast) [reply README.md:28]: Ah, this was an accident, will re-add!

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hello/db/DbController.kt (hunk 4 above)
Comment (Silvast) [reply services/hanke-service/src/main/kotlin/fi/hel/haitaton/hello/db/DbController.kt:46]: thanks, I fixed this.

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hello/db/DbController.kt (hunk 6 above)
Comment (Silvast) [reply services/hanke-service/src/main/kotlin/fi/hel/haitaton/hello/db/DbController.kt:28]: ah, hyvä pointti, kiitos!

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hello/db/DbController.kt (hunk 3 above)
Comment (Silvast) [reply services/hanke-service/src/main/kotlin/fi/hel/haitaton/hello/db/DbController.kt:30]: Yes, datasource might be better. The problem there was that if I deternmine spring.datasource to the application.properties, it connects automatically and running fails with no db. So, I didn't want the starting of the app to fail if no db is yet available (as is the situation in our cloud service at the moment). But yeah, definately we should re-check this when we know better..

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hello/db/DbController.kt (hunk 2 above)
Comment (Silvast) [reply services/hanke-service/src/main/kotlin/fi/hel/haitaton/hello/db/DbController.kt:47]: Ah, you are right, thanks for commenting, I will fixed this to close as well!

**Pull request merged**